### PR TITLE
Deploy and upgrade operator via OLM

### PR DESCRIPTION
**Pull request by nitishkumar4**

### CODE REVIEW

File: generated/installer/ibm-spectrum-scale-csi-operator-dev.yaml [hunk 1]
```diff
@@ -46,7 +48,7 @@ spec:
               fieldPath: metadata.namespace
         - name: CSI_DRIVER_IMAGE
           value: quay.io/ibm-spectrum-scale-dev/ibm-spectrum-scale-csi-driver:dev
-        image: quay.io/ibm-spectrum-scale-dev/ibm-spectrum-scale-csi-operator:dev
+        image: quay.io/ibm-spectrum-scale/ibm-spectrum-scale-csi-operator:v2.5.0
```
Comment (madhuthorat) [inline generated/installer/ibm-spectrum-scale-csi-operator-dev.yaml]: This needs to be "dev" image too!

File: operator/config/olm-catalog/ibm-spectrum-scale-csi-operator/2.5.0/ibm-spectrum-scale-csi-operator.v2.5.0.clusterserviceversion.yaml [hunk 2]
```diff
@@ -1,295 +1,342 @@
...
+          "status": {}
+        }
+      ]
     capabilities: Basic Install
     categories: Storage
-    certified: 'false'
+    certified: "false"
     containerImage: quay.io/ibm-spectrum-scale/ibm-spectrum-scale-csi-operator:v2.5.0
-    createdAt: Mon Dec 20 14:55:05 IST 2021
+    createdAt: Mon Nov 01 13:55:05 IST 2021
```
Comment (madhuthorat) [inline operator/config/olm-catalog/ibm-spectrum-scale-csi-operator/2.5.0/ibm-spectrum-scale-csi-operator.v2.5.0.clusterserviceversion.yaml]: Date needs to be Mon Dec 20 14:55:05 IST 2021

File: operator/config/olm-catalog/ibm-spectrum-scale-csi-operator/2.5.0/ibm-spectrum-scale-csi-operator.v2.5.0.clusterserviceversion.yaml [hunk 3]
```diff
@@ -1,295 +1,342 @@
...
     description: An operator for deploying and managing the IBM Spectrum Scale CSI
       Driver.
-    operators.operatorframework.io/builder: operator-sdk-v1.15.0
-    operators.operatorframework.io/project_layout: go.sdk.operatorframework.io/v1
+    operators.operatorframework.io/builder: operator-sdk-v1.14.0+git
```
Comment (madhuthorat) [inline operator/config/olm-catalog/ibm-spectrum-scale-csi-operator/2.5.0/ibm-spectrum-scale-csi-operator.v2.5.0.clusterserviceversion.yaml:77]: We need to use latest operator sdk 1.15.0

File: operator/config/manifests/bases/ibm-spectrum-scale-csi-operator.clusterserviceversion.yaml [hunk 4]
```diff
@@ -21,277 +29,260 @@ metadata:
...
+        path: clusters[0].secrets
         x-descriptors:
-        - urn:alm:descriptor:com.tectonic.ui:label
-      - description: Require a secure SSL connection to connect to IBM Spectrum Scale  REST server
+        - urn:alm:descriptor:io.kubernetes:Secret
+      - description: secureSslMode specifies if a secure SSL connection to connect
+          to GPFS is required.
         displayName: Secure SSL Mode
-        path: clusters.secureSslMode
+        path: clusters[0].secureSslMode
```
Comment (madhuthorat) [inline operator/config/manifests/bases/ibm-spectrum-scale-csi-operator.clusterserviceversion.yaml:110]: Will this work for remote clusters?

File: generated/installer/ibm-spectrum-scale-csi-operator-dev.yaml (hunk 1 above)
Comment (nitishkumar4) [reply generated/installer/ibm-spectrum-scale-csi-operator-dev.yaml]: fixed

File: operator/config/olm-catalog/ibm-spectrum-scale-csi-operator/2.5.0/ibm-spectrum-scale-csi-operator.v2.5.0.clusterserviceversion.yaml (hunk 2 above)
Comment (nitishkumar4) [reply operator/config/olm-catalog/ibm-spectrum-scale-csi-operator/2.5.0/ibm-spectrum-scale-csi-operator.v2.5.0.clusterserviceversion.yaml]: fixed

File: operator/config/olm-catalog/ibm-spectrum-scale-csi-operator/2.5.0/ibm-spectrum-scale-csi-operator.v2.5.0.clusterserviceversion.yaml (hunk 3 above)
Comment (nitishkumar4) [reply operator/config/olm-catalog/ibm-spectrum-scale-csi-operator/2.5.0/ibm-spectrum-scale-csi-operator.v2.5.0.clusterserviceversion.yaml:77]: have latest version in my environment, `operator-sdk version: "v1.15.0", commit: "f6326e832a8a5e5453d0ad25e86714a0de2c0fc8", kubernetes version: "1.21", go version: "go1.16.10", GOOS: "linux", GOARCH: "amd64"`

File: operator/config/manifests/bases/ibm-spectrum-scale-csi-operator.clusterserviceversion.yaml (hunk 4 above)
Comment (nitishkumar4) [reply operator/config/manifests/bases/ibm-spectrum-scale-csi-operator.clusterserviceversion.yaml:110]: yes, it should have not impact on remotecluster, though the controller code is not fully capable to handle multiple values under spec.cluster as I can see error in operator logs.

Comment (nitishkumar4) [reply operator/config/manifests/bases/ibm-spectrum-scale-csi-operator.clusterserviceversion.yaml:110]: If only remotecluster details are passed, it should be just fine. Tested it.

Comment (nitishkumar4) [reply operator/config/manifests/bases/ibm-spectrum-scale-csi-operator.clusterserviceversion.yaml:110]: Regarding path field, this used by OCP UI to display details from CR in UI, this is implemented correctly now. in previous versions this was not implemented correctly as I can see in older builds 2.4.0 UI in not showing CR details correctly(missing).

**Review by madhuthorat — APPROVED** [review]

**Pull request merged**